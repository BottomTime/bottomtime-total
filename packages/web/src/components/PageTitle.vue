<template>
  <section class="hero is-dark">
    <div class="hero-body">
      <div class="container">
        <p class="title">{{ title }}</p>
        <p v-if="subtitle" class="subtitle">{{ subtitle }}</p>
        <slot></slot>
      </div>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { onMounted } from 'vue';

interface PageTitleProps {
  title: string;
  subtitle?: string;
}

const props = defineProps<PageTitleProps>();

onMounted(() => {
  document.title = `BottomTime | ${props.title}`;
});
</script>
