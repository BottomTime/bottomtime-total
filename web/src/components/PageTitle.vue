<template>
  <section class="hero is-info">
    <div class="hero-body">
      <div class="container">
        <p class="title">{{ title }}</p>
        <p v-if="subtitle" class="subtitle">{{ subtitle }}</p>
      </div>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { defineProps, onMounted } from 'vue';

interface PageTitleProps {
  title: string;
  subtitle?: string;
}

const props = defineProps<PageTitleProps>();

onMounted(() => {
  document.title = `BottomTime | ${props.title}`;
});
</script>
