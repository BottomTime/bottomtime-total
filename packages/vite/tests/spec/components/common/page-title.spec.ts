import { mount } from '@vue/test-utils';
import PageTitle from '../../../../src/components/common/page-title.vue';

describe('Page Title component', () => {
  it('will render correctly with subtitle', () => {
    const wrapper = mount(PageTitle, {
      props: {
        title: 'Test Title',
        subtitle: 'Test Subtitle',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('will render correctly without subtitle', () => {
    const wrapper = mount(PageTitle, {
      props: {
        title: 'Test Title',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
