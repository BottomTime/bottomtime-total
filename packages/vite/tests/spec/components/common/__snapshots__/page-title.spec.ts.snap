// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Title component will render correctly with subtitle 1`] = `
"<div class="flex-none h-30 pt-2 pb-7 font-title text-blue-950">
  <p class="text-6xl mb-6">Test Title</p>
  <p class="text-2xl">Test Subtitle</p>
</div>"
`;

exports[`Page Title component will render correctly without subtitle 1`] = `
"<div class="flex-none h-30 pt-2 pb-7 font-title text-blue-950">
  <p class="text-6xl mb-6">Test Title</p>
  <!--v-if-->
</div>"
`;
